agent/main.go

```diff
@@ -226,6 +229,28 @@ func main() {
 		return
 	}
 
+	// if secure-access is enabled create csr
+	if feature.Gates.Enabled(feature.SecureAccess) {
+		logger.Info("creating host csr")
+		byohCSR := registration.ByohCSR{K8sClient: k8sClient}
+		// TODO: persist this key across host-agent restarts
+		privKey, err := byohCSR.CreateCSR(hostName)
+		if err != nil {
+			logger.Error(err, "host csr creation failed")
+			return
```

Comment:
Is this not a stop-the-agent/world kind of error?

Reply:
This is, as for the bootstrap-kubeconfig flow, we first need to create a CSR to proceed further with the other flows which will use generated certificate for this CSR.

Reply:
Created this issue #520 for proper handling of exit codes.